### PR TITLE
[IMAP Sensor] Support for automatically downloading and storing message attachments in the datastore

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds support for automatically downloading and store message attachments in the datastore.

This allows users to retrieve those attachments from a datastore later (e.g. inside an action).

For more information on how it works, pleasae refer to the readme.

### CODE REVIEW

File: packs/email/README.md [hunk 1]
```diff
@@ -35,4 +36,150 @@ imap_mailboxes:
...
+    download_attachments: True
+max_attachment_size: 1024
+attachment_datastore_ttl: 1800
+```
+
+The following attachment settings can be configured:
+
+* ``download_attachments`` - True to download the attachment and store them in the
+  datastore.
+* ``max_attachment_size`` - Maximum size of download attachment. If an attachment
```
Comment (manasdk) [inline packs/email/README.md]: unit?

File: packs/email/sensors/imap_sensor.py [hunk 2]
```diff
@@ -101,7 +135,57 @@ def _process_message(self, uid, mailbox):
...
+
         self._sensor_service.dispatch(trigger=self._trigger, payload=payload)
+
+    def _download_and_store_message_attachments(self, message):
+        """
+        Method which downloads the provided message attachments and stores them in a datasatore.
+
+        :rtype: ``list`` of ``dict``
+        """
+        attachments = message.attachments
```
Comment (manasdk) [inline packs/email/sensors/imap_sensor.py]: So the attachment is downloaded regardless of size anyway - correct?

We discard the attachment if it is larger than configured but will be in memory AFAICT.

**Comment by manasdk** [comment]
LGTM

File: packs/email/sensors/imap_sensor.py [hunk 3]
```diff
@@ -35,13 +47,17 @@ def setup(self):
     def poll(self):
         self._logger.debug('[IMAPSensor]: entering poll')
 
-        for name, mailbox in self._mailboxes.items():
-            self._poll_for_unread_messages(name, mailbox)
+        for name, values in self._mailboxes.items():
```
Comment (manasdk) [inline packs/email/sensors/imap_sensor.py]: six?

File: packs/email/README.md (hunk 1 above)
Comment (Kami) [inline packs/email/README.md]: Clarified units in the readme.

File: packs/email/sensors/imap_sensor.py (hunk 2 above)
Comment (Kami) [inline packs/email/sensors/imap_sensor.py]: Yeah, sadly that's how the underlying library works - the attachment is still downloaded, but we don't store it in the datastore.

Not ideal, but we can look into optimizing it (not downloading the attachment) later.

File: packs/email/sensors/imap_sensor.py (hunk 3 above)
Comment (Kami) [inline packs/email/sensors/imap_sensor.py]: We don't consistently use `six` in st2contrib, we should probably consider adding `six` to the base requirements list which are installed with every pack.

File: packs/email/sensors/imap_sensor.py (hunk 2 above)
Comment (Kami) [inline packs/email/sensors/imap_sensor.py]: Clarified that in the readme.

**Pull request merged**